driver/docker/version.go

```diff
@@ -0,0 +1,38 @@
+package docker
+
+import (
+	_ "embed"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+// https://github.com/Masterminds/semver#checking-version-constraints
+var mobyBuildkitVersions = map[string]string{
+	"18.06.0-0 - 18.09.0-0": "v0.3.0",
+	"18.09.1-0 - 18.09.9-0": "v0.4.0",
+	"19.03.0-0 - 19.03.1-0": "v0.6.0",
+	"19.03.2-0 - 19.03.6-0": "v0.6.2",
+	"~19.03.x-0":            "v0.6.4",
+	"20.10.0-0 - 20.10.3-0": "v0.8.1",
+	"~20.10.x-0":            "v0.8.2",
+	"~22.06.x-0":            "v0.10.3",
+	"~23.0.0-0":             "v0.10.6",
+	"~23":                   "v0.10.6",
+	// no need to have a constraint for 24+ as the API already returns the
+	// right BuildKit version: https://github.com/moby/moby/pull/44917
+}
```

Comment:
Switch [version constraint](https://github.com/Masterminds/semver#checking-version-constraints) instead with hyphen and tilde ranges comparisons.

@thaJeztah Not quite sure for `~23: v0.10.6`. Maybe it should be `v0.10+unknown`?

Reply:
Same for `~20.10.x-0: v0.8.2`.

We could have:
* `20.10.0: v0.8.1`: https://github.com/moby/moby/blob/v20.10.0/vendor.conf#L36
* `20.10.1: v0.8.1`: https://github.com/moby/moby/blob/v20.10.1/vendor.conf#L36
* `20.10.2: v0.8.1`: https://github.com/moby/moby/blob/v20.10.2/vendor.conf#L36
* `20.10.3: v0.8.2-1-g68bb095`: https://github.com/moby/moby/blob/v20.10.3/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/68bb095353c65bc3993fd534c26cf77fe05e61b1
* `20.10.4: v0.8.2`: https://github.com/moby/moby/blob/v20.10.4/vendor.conf#L36
* `20.10.5: v0.8.2`: https://github.com/moby/moby/blob/v20.10.5/vendor.conf#L36
* `20.10.6: v0.8.2`: https://github.com/moby/moby/blob/v20.10.6/vendor.conf#L36
* `20.10.7: v0.8.3-3-g244e8cde`: https://github.com/moby/moby/blob/v20.10.7/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/244e8cde639f71a05a1a2e0670bd88e0206ce55c
* `20.10.8: v0.8.3-3-g244e8cde`: https://github.com/moby/moby/blob/v20.10.8/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/244e8cde639f71a05a1a2e0670bd88e0206ce55c
* `20.10.9: v0.8.3-3-g244e8cde`: https://github.com/moby/moby/blob/v20.10.9/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/244e8cde639f71a05a1a2e0670bd88e0206ce55c
* `20.10.10: v0.8.3-3-g244e8cde`: https://github.com/moby/moby/blob/v20.10.10/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/244e8cde639f71a05a1a2e0670bd88e0206ce55c
* `20.10.11: v0.8.3-4-gbc07b2b8`: https://github.com/moby/moby/blob/v20.10.11/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/bc07b2b81b1c6a62d29981ac564b16a15ce2bfa7
* `20.10.12: v0.8.3-4-gbc07b2b8`: https://github.com/moby/moby/blob/v20.10.12/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/bc07b2b81b1c6a62d29981ac564b16a15ce2bfa7
* `20.10.13: v0.8.3-4-gbc07b2b8`: https://github.com/moby/moby/blob/v20.10.13/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/bc07b2b81b1c6a62d29981ac564b16a15ce2bfa7
* `20.10.14: v0.8.3-4-gbc07b2b8`: https://github.com/moby/moby/blob/v20.10.14/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/bc07b2b81b1c6a62d29981ac564b16a15ce2bfa7
* `20.10.15: v0.8.3-4-gbc07b2b8`: https://github.com/moby/moby/blob/v20.10.15/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/bc07b2b81b1c6a62d29981ac564b16a15ce2bfa7
* `20.10.16: v0.8.3-4-gbc07b2b8`: https://github.com/moby/moby/blob/v20.10.16/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/bc07b2b81b1c6a62d29981ac564b16a15ce2bfa7
* `20.10.17: v0.8.3-4-gbc07b2b8`: https://github.com/moby/moby/blob/v20.10.17/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/bc07b2b81b1c6a62d29981ac564b16a15ce2bfa7
* `20.10.18: v0.8.3-4-gbc07b2b8`: https://github.com/moby/moby/blob/v20.10.18/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/bc07b2b81b1c6a62d29981ac564b16a15ce2bfa7
* `20.10.19: v0.8.3-4-gbc07b2b8`: https://github.com/moby/moby/blob/v20.10.19/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/bc07b2b81b1c6a62d29981ac564b16a15ce2bfa7
* `20.10.19: v0.8.3-29-g3a1eeca5`: https://github.com/moby/moby/blob/v20.10.19/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/3a1eeca59a9263613d996ead67d53a4b7d45723d
* `20.10.20: v0.8.3-31-gc0149372`: https://github.com/moby/moby/blob/v20.10.20/vendor.conf#L36
  * https://github.com/moby/buildkit/commit/c014937225cba29cfb1d5161fd134316c0e9bdaa
* `20.10.21: v0.8.3-34-geeb7b65`: https://github.com/moby/moby/blob/v20.10.21/vendor.conf#L37
  * https://github.com/moby/buildkit/commit/eeb7b65ab7d651770a5ec52a06ea7c96eb97a249
* `20.10.22: v0.8.3-34-geeb7b65`: https://github.com/moby/moby/blob/v20.10.22/vendor.conf#L37
  * https://github.com/moby/buildkit/commit/eeb7b65ab7d651770a5ec52a06ea7c96eb97a249
* `20.10.23: v0.8.3-34-geeb7b65`: https://github.com/moby/moby/blob/v20.10.23/vendor.conf#L37
  * https://github.com/moby/buildkit/commit/eeb7b65ab7d651770a5ec52a06ea7c96eb97a249